web/packages/teleterm/src/ui/TopBar/Identity/Identity.tsx

```diff
@@ -68,109 +67,77 @@ export function IdentityContainer() {
       'openProfiles'
     );
 
+  function withClose<T extends (...args: any[]) => any>(
+    fn: T
+  ): (...args: Parameters<T>) => ReturnType<T> {
+    return (...args) => {
+      setOpen(false);
+      return fn(...args);
+    };
+  }
+
+  const deviceTrustStatus = calculateDeviceTrustStatus(
+    activeRootCluster?.loggedInUser
+  );
+  const activeColor = useStoreSelector(
+    'workspacesService',
+    useCallback(
+      state =>
+        activeRootCluster
+          ? state.workspaces[activeRootCluster.uri]?.color
+          : undefined,
+      [activeRootCluster]
```

Comment:
This could be refactored to not depend on `activeRootCluster`.

```typescript
  const activeColor = useStoreSelector(
    'workspacesService',
    useCallback(state => state.workspaces[state.rootClusterUri]?.color, [])
  );
```

I'm yet to made up my mind which way is better. But the one that doesn't depend on `activeRootCluster` doesn't ever change its subscription callback, so I suppose it's better in that way?

Reply:
Yeah, with the current version, the benefits of `useStoreSelector` are pretty much gone. When you change the active workspace, `activeColor` gets updated not because the correct part of `workspacesService` is updated, but because `activeRootCluster` has changed. The callback passed to `useStoreSelector` always returns the same color (until `activeRootCluster` changes).

Reply:
I don't know why I thought of putting the entire `activeRootCluster ` object as a dependency in the store selector. Replacing it with `state.rootClusterUri` makes much more sense, thanks!